File: src/main/kotlin/com/io/reactivecoroutines/weather/api/WeatherAPIResponse.java
package com.io.reactivecoroutines.weather.api;

import com.io.reactivecoroutines.weather.WeatherInfo;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public record WeatherAPIResponse(@NotNull Location location, @NotNull Forecast forecast) {
    public List<WeatherInfo> toWeatherInfoList() {
        final var region = location.region();
        final var country = location.country();
        final var city = location.city();
        return forecast.days().stream().map(f ->
                new WeatherInfo(
                        region,
                        country,
                        null,
                        city,
                        f.date(),
                        "%.2f".formatted(f.temperature().avgF())
                )
        ).toList();
    }
}

File: src/main/kotlin/com/io/reactivecoroutines/weather/api/WeatherAPIResponse.kt
package com.io.reactivecoroutines.weather.api

import com.io.reactivecoroutines.weather.WeatherInfo

data class WeatherAPIResponse(val location: Location, val forecast: Forecast) {
    fun toWeatherInfoList(): List<WeatherInfo> {
        return forecast.days.stream().map { f ->
            with(location) {
                WeatherInfo(
                    id = null,
                    region = region,
                    country = country,
                    state = null,
                    city = city,
                    localDate = f.date,
                    avgTemperature = "${f.temperature.avgF}"
                )
            }
        }.toList()
    }
}